src/pretix/control/views/subevents.py

```diff
@@ -184,11 +184,26 @@ def delete(self, request, *args, **kwargs):
                                                               'placed.'))
             return HttpResponseRedirect(self.get_success_url())
         else:
-            self.object.log_action('pretix.subevent.deleted', user=self.request.user)
-            CartPosition.objects.filter(addon_to__subevent=self.object).delete()
-            self.object.cartposition_set.all().delete()
-            self.object.delete()
-            messages.success(request, pgettext_lazy('subevent', 'The selected date has been deleted.'))
+            try:
+                self.object.log_action('pretix.subevent.deleted', user=self.request.user)
+                CartPosition.objects.filter(addon_to__subevent=self.object).delete()
+                self.object.cartposition_set.all().delete()
+                self.object.delete()
+                messages.success(request, pgettext_lazy('subevent', 'The selected date has been deleted.'))
```

Comment:
Hm, it feels a bit strange to me to have a deleted-log-entry shown to me when editing a failed-to-be-deleted subevent. I know it means, that someone attempted to delete the subevent, but still feels weird. Also on the eventseries-dashboard, the log-entry shows up right before an changed-log for the same subevent. I think we should log the delete only if it succeeds. So:

```suggestion
                CartPosition.objects.filter(addon_to__subevent=self.object).delete()
                self.object.cartposition_set.all().delete()
                self.object.delete()
                self.object.log_action('pretix.subevent.deleted', user=self.request.user)
                messages.success(request, pgettext_lazy('subevent', 'The selected date has been deleted.'))
```

Reply:
~I don't think we need a delete_Failed, but reversing the order makes total sense~ is what I thought, but "delete first, log second" is not possible because then self.object no longer has an ID that the log can be stored with

Reply:
Oh, makes total sense. Never mind, then it’s fine like it is!